core/src/test/java/edu/ucr/cs/riple/core/tools/CoreTestHelper.java

```diff
@@ -198,16 +198,17 @@ private void compare(Collection<Report> actualOutput) {
 
   private void makeAnnotatorConfigFile(Path path) {
     Config.Builder builder = new Config.Builder();
+    final int[] id = {0};
     builder.configPaths =
         modules.stream()
             .map(
                 name ->
                     new ModuleInfo(
+                        id[0]++,
```

Comment:
side effect inside a lambda :|

Reply:
Streams and lambdas in Java are not meant to be pure, though. There is a reason the `.forEach(...)` operator exists, which would be pointless in a functional language. Obviously it is a 'use this with care' situation, but incrementing a counter in a sequential stream is not an unreasonable thing to do (or even in a parallel stream, though that would need an atomic counter 🙂 ).